src/drt/src/TritonRoute.cpp

```diff
@@ -976,6 +953,65 @@ void TritonRoute::checkDRC(const char* filename, int x1, int y1, int x2, int y2)
   reportDRC(filename, markers, requiredDrcBox);
 }
 
+void TritonRoute::stackVias(odb::dbBTerm* bterm,
+                            int top_layer_idx,
+                            int bterm_bottom_layer_idx)
+{
+  odb::dbTech* tech = db_->getTech();
+  odb::dbSet<odb::dbTechVia> vias = tech->getVias();
+  std::map<int, odb::dbTechVia*> default_vias;
+  for (int i = top_layer_idx; i <= bterm_bottom_layer_idx; i++) {
+    for (odb::dbTechVia* via : vias) {
+      if (via->getBottomLayer()->getRoutingLevel() == i) {
+        default_vias[i] = via;
+        break;
+      }
+    }
+  }
+
+  // get bterm access points
+  std::vector<odb::dbAccessPoint*> access_points;
+  odb::Rect pin_rect;
+  for (odb::dbBPin* bpin : bterm->getBPins()) {
+    const std::vector<odb::dbAccessPoint*>& bpin_pas = bpin->getAccessPoints();
+    access_points.insert(
+        access_points.begin(), bpin_pas.begin(), bpin_pas.end());
+    pin_rect = bpin->getBBox();
+  }
+
+  // set the via position as the first AP in the same layer of the bterm
+  odb::dbTechLayer* bterm_bottom_tech_layer
+      = tech->findRoutingLayer(bterm_bottom_layer_idx);
+  odb::Point via_position;
+  if (!access_points.empty()) {
+    for (const auto& ap : access_points) {
+      if (ap->getLayer()->getRoutingLevel()
+          == bterm_bottom_tech_layer->getRoutingLevel()) {
+        via_position = ap->getPoint();
+      }
+    }
+  } else {
+    via_position = odb::Point(pin_rect.xCenter(), pin_rect.yCenter());
+  }
+
+  // insert the vias from the top routing layer to the bterm bottom layer
+  odb::dbNet* net = bterm->getNet();
+  odb::dbWire* wire = net->getWire();
+  if (wire == nullptr) {
```

Comment:
If there is an existing route then this is dangerous as you are erasing it.  It seems like an error given the intended use model.